sdk/eventgrid/Azure.Messaging.EventGrid/EventGridSourceGenerator/EventGridSourceGenerator.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Text;
+
+namespace EventGridSourceGenerator
+{
+    [Generator]
+    public class EventGridSourceGenerator : ISourceGenerator
+    {
+        private SourceVisitor _visitor;
+        private const string Indent = "    ";
+
+        public void Execute(GeneratorExecutionContext context)
+        {
+            // using the context, get a list of syntax trees in the users compilation
+            var syntaxTrees = context.Compilation.WithAssemblyName("Azure.Messaging.EventGrid").SyntaxTrees;
+            _visitor = new SourceVisitor();
+
+            foreach (CSharpSyntaxTree tree in syntaxTrees)
+            {
+                _visitor.Visit(tree.GetCompilationUnitRoot());
+            }
+
```

Comment:
Surprised the linter didn't bug you about multiple blank lines here?

Reply:
Yeah, I think it is not enabled because this isn't a IsShippingClientLibrary. I tried manually adding the analyzers and still couldn't get it to work though.